### PR TITLE
chore: Add checkstyle

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**
Added CheckStyle to have control over code formatting. All subprojects
and included builds will use it.

APM-2638

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,6 @@ captures/
 
 # Output folder of infer static analysis tool: http://fbinfer.com/
 infer-out/
+
+# Jacoco files
+jacoco.exec
```
Comment (richard-bogdan-bitrise) [inline .gitignore:35]: Note: this change is not related to the rest of the changes

File: checkstyle.gradle [hunk 2]
```diff
@@ -0,0 +1,68 @@
+checkstyle {
+    configFile file("../checkstyle.xml")
+    ignoreFailures false
+    showViolations false
+    toolVersion = "8.42"
+}
+
+if (project.hasProperty("android")) {
```
Comment (richard-bogdan-bitrise) [inline checkstyle.gradle:8]: Android vs Java modules.

File: checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,351 @@
+<?xml version="1.0"?><!DOCTYPE module PUBLIC
```
Comment (richard-bogdan-bitrise) [inline checkstyle.xml:1]: Based on [Google Java Code Style](https://google.github.io/styleguide/javaguide.html).

**Comment by richard-bogdan-bitrise** [comment]
Decided to go with [CheckStyle](urlhttps://checkstyle.sourceforge.io), as I wanted to go with a well known code formatting rules, for example Google's Java code formatting rules. CheckStyle supports it.

File: build.gradle [hunk 4]
```diff
@@ -107,3 +107,21 @@ sonarqube {
         }
     }
 }
+
+/**
+ * Task that will call the checkStyle task fpr all subproject and included build.
```
Comment (daniellevass-bitrise) [inline build.gradle:112]: nit: fpr -> for?

What do you mean the checkstyle will be called for the included build? can you also explain why this is needed?

File: build.gradle [hunk 5]
```diff
@@ -107,3 +107,21 @@ sonarqube {
...
+
+    project.getSubprojects().forEach {
+        it.afterEvaluate {
+            dependsOn it.tasks.findByName('checkStyle')
+        }
+    }
+}
+
+
+
```
Comment (daniellevass-bitrise) [inline build.gradle:127]: nit: extra lines at the end.

File: checkstyle.xml (hunk 3 above)
Comment (daniellevass-bitrise) [reply checkstyle.xml:1]: Could you maybe update the android confluence docs to include this?

**Review by daniellevass-bitrise — APPROVED** [review]
Left a few nits, but the rest looks great 🚀 

Could you also update the description to give an example of what running this task locally produces? what does an acceptable build look like vs a build with some failures? This is super useful for us to understand how this task *should* run, and we then have a datapoint if anything changes in the future.

File: checkstyle.xml (hunk 3 above)
Comment (richard-bogdan-bitrise) [reply checkstyle.xml:1]: Will definitely dp it, it will be the last subtask for this.

File: build.gradle (hunk 4 above)
Comment (richard-bogdan-bitrise) [reply build.gradle:112]: Thanks for spotting it, I also spotted it later, and fixed it with this [commit](https://github.com/bitrise-io/trace-android-sdk/pull/204/commits/735fccfdd53c4cd30afc0d2e4206327c509af34a) if it is ok.

Currently trace-android-sdk have 2 subprojects, trace-sdk and trace-test-applicaiton, and 3 included builds, buildSrc, pencensus and trace-gradle-plugin. 
The purpose of this is to be an umbrella task, that will call checkStyle tasks in all subprojects and included builds.

File: build.gradle (hunk 5 above)
Comment (richard-bogdan-bitrise) [reply build.gradle:127]: Removed them.

File: checkstyle.xml (hunk 3 above)
Comment (daniellevass-bitrise) [reply checkstyle.xml:1]: Are you still planning to add this to the confluence doc? I noticed you edited the page and added the other links, but this would be useful to add too :)

Comment (richard-bogdan-bitrise) [reply checkstyle.xml:1]: It is on the checkstlye page, but I can add the direct link for it.

**Comment by richard-bogdan-bitrise** [comment]
Rebased to main.
